CONTRIBUTING.md

```diff
@@ -17,3 +17,56 @@ The contributor roles for the new CDK construct library initiative are designed
 **Maintainers** are responsible for overseeing the entire project. They make important decisions, coordinate efforts, and ensure the overall quality and direction of the library. Maintainers are chosen among reviewers that have shown a good grasp around the overall direction of the project.
 
 Outside of these contributor roles, there are additional roles of involvement within the Open Construct Foundation. These roles include board members and officers who oversee the organization and its initiatives. Board members provide strategic guidance and uphold the Foundation's mission. Officers decide on the direction of the foundation based on the advise from the board members and are responsible for specific functional areas like finance, operations, or community engagement.
+
+## Construct Design
+
+Construct design in this library has two primary goals:
+
+1. Maintain a high level of consistency across constructs to make them easy to use and understand.
+2. Ensure that little to no changes are required to merge constructs into the core AWS CDK library.
+
+To achieve these goals, the following design principles are followed:
+
+* Focus on only accepting L2 constructs to start. L3 constructs will come later as the library matures.
+* Follow the AWS CDK contributing guides
+* Use the AWS CDK library as a reference for design patterns
```

Comment:
Should we write a link to the design guidelines for aws-cdk?

https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md